File: app/src/at-trivo/java/asiantech/internship/summer/recycler_view/TimelineViewHolder.java
package asiantech.internship.summer.recycler_view;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import asiantech.internship.summer.R;

public class TimelineViewHolder extends RecyclerView.ViewHolder {

    private final ImageView mImgAuthor;
    private final TextView mTvAuthorName;
    private final ImageView mImgTimeline;
    private final TextView mTvLoveNumber;
    private final TextView mTvDescription;

    private Integer mPosition;
    private TimeLineViewHolderListener mListener;

    public void setPosition(int position){
        this.mPosition=position;
    }

    public ImageView getImgAuthor() {
        return mImgAuthor;
    }

    public TextView getTvAuthorName() {
        return mTvAuthorName;
    }

    public ImageView getImgTimeline() {
        return mImgTimeline;
    }

    public TextView getTvLoveNumber() {
        return mTvLoveNumber;
    }

    public TextView getTvDescription() {
        return mTvDescription;
    }

    public TimelineViewHolder(View itemView,TimeLineViewHolderListener listener) {
        super(itemView);
        mListener=listener;
        mImgAuthor=itemView.findViewById(R.id.imgAuthor);
        mTvAuthorName=itemView.findViewById(R.id.tvAuthorName);
        mImgTimeline=itemView.findViewById(R.id.imgTimeline);
        mTvLoveNumber=itemView.findViewById(R.id.tvLoveNumber);
        mTvDescription=itemView.findViewById(R.id.tvDescription);

        ImageView imgHeart=itemView.findViewById(R.id.imgHeart);
        imgHeart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mListener.onHeartImageClick(mPosition);
            }
        });
    }



    public interface TimeLineViewHolderListener{
        void onHeartImageClick(int position);
    }
}

File: app/src/at-trivo/java/asiantech/internship/summer/recycler_view_kotlin/timeline_recycler_view/TimelineViewHolder.kt
package asiantech.internship.summer.recycler_view_kotlin.timeline_recycler_view

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import asiantech.internship.summer.R
import asiantech.internship.summer.recycler_view_kotlin.TimelineFragmentListener

class TimelineViewHolder(itemView: View?, val listener: TimelineFragmentListener) : RecyclerView.ViewHolder(itemView) {
    val tvAuthorName: TextView = itemView!!.findViewById(R.id.tvAuthorName)
    val imgAuthor: ImageView = itemView!!.findViewById(R.id.imgAuthor)
    val tvDescription: TextView = itemView!!.findViewById(R.id.tvDescription)
    val imgTimeline: ImageView = itemView!!.findViewById(R.id.imgTimeline)
    val imgHeart: ImageView = itemView!!.findViewById(R.id.imgHeart)

    init {
        imgHeart.setOnClickListener {
            listener.onHeartImageClicked(dataPosition)
        }
    }

    var dataPosition: Int = 0
    val tvLoveNumber: TextView = itemView!!.findViewById(R.id.tvLoveNumber)
}